test/cases/ref.ts

```diff
@@ -0,0 +1,40 @@
+export var schema = 
+{
+  "title": "Referencing",
+  "type": "object",
+  "properties": {
+    "foo": {
+      "$ref": "test/resources/ReferencedType.json"
+    }
+  },
+  "required": ["foo"],
+  "additionalProperties": false
+}
+
+export var configurations = [
+  {
+    settings: {
+      declareReferenced: true
```

Comment:
I would rather not bloat the API unless it's broadly useful. If you don't mind, let's remove it and add a batch mode in the future. Since these are generated files anyway, the bit of clutter won't matter much.